### ISSUE TITLE
Roadmap to 2.0

### ISSUE THREAD

**Issue by otakustay** [otakustay/react-diff-view#8]
Currentnt `1.x` version works quite well, however with some little inconveniences, the most significant is about the hunk header implementation

Suppose we have a quite complex code expanding UI like:

<img width="1145" alt="jietu20171215-151055" src="https://user-images.githubusercontent.com/639549/34031083-add8c98a-e1aa-11e7-8fc6-4cf8f95b593e.png">

Visually we will think it should be:

<img width="1145" alt="jietu20171215-151055-1" src="https://user-images.githubusercontent.com/639549/34031088-b7bc8108-e1aa-11e7-9a55-00bc517e6d2c.png">

(This is not very correct, the last rectangle is wrong)

In fact, with the limitation that each hunk can have only one header, the actual structure is:

<img width="1145" alt="jietu20171215-151055-2" src="https://user-images.githubusercontent.com/639549/34031106-cf07331c-e1aa-11e7-8aae-2f6b65701e57.png">

This is **weird** and spiritually **intolerable**

To solve this problem, the `2.x` version will have a better structure, in which we allow any count of any elements before, after and between hunks, we make it a `Decoration` component

A `Decoration` component receives only a `children` prop which can either be one or two react elements, `Decoration` will wrap them into a `<tr>` element with the same logic of current `Hunk.header` prop

In this way, we can simply write:

```jsx
<Diff>
    <Decoration>
        <Unfold start={1} end={...}>Expand all</Unfold>
    </Decoration>
    <Hunk hunk={...} />
    <Decoration>
        <Unfold ... />
    </Decoration>>
    <Decoration>
        <Unfold ... />
    </Decoration>>
    <Decoration>
        <Unfold ... />
    </Decoration>
    <Hunk hunk={...} />
</Diff>
```

In order to archive this, our tasks are:

- Implement `Decoration` component
- Allow `Diff` component to receive `Decoration` as children
- Remove header related props in `customEvents`
- Further remove all currently deprecated features

**Comment by VulumeCode** [comment]
Why put the master (frontpage readme, default branch for clone) at a higher major version? 

This lost me a day trying to get the demo to work with the npm package... Please avoid confusion.

**Comment by otakustay** [comment]
Sorry for the inconvenience, I'm too anxious in the 2.0 release, I'll keep this in mind in later major versions.

Just released `2.0.1` on the latest tag which is exactly the same as `2.0.0-alpha` only updating react's peer version to `>= 16.8`

Will work on hooks to release `2.0.2` later
